WcaOnRails/app/webpacker/components/CompetitionOverview/QueryHelper.js

```diff
@@ -0,0 +1,69 @@
+const isContinent = (region) => region[0] === '_';
+
+export function calculateQueryKey(filterState) {
+  let timeKey = '';
+  if (filterState?.timeOrder === 'past') {
+    timeKey = `${filterState.selectedYear}`;
+  } else if (filterState?.timeOrder === 'custom') {
+    timeKey = `start${filterState.customStartDate}-end${filterState.customEndDate}`;
+  }
+
+  return {
+    timeOrder: filterState?.timeOrder,
+    region: filterState?.region,
+    delegate: filterState?.delegate,
+    search: filterState?.search,
+    time: timeKey,
+  };
+}
+
+export function createSearchParams(filterState, pageParam) {
```

Comment:
Suggestion: you could get rid of all the `filterState.`s by adding `const { region, delegate, search, [...etc] } = filterState` at the top of this function. It's just up to personal preference though.
ie the first `if` below becomes `if (region && region !== 'all_regions')`